src/lib.rs

```diff
@@ -1,8 +1,530 @@
+                            kind.to_upper_camel_case(),
+                        )?;
+                    }
+
+                    if self.schema_mode != SchemaMode::Derived {
+                        writeln!(&mut generated, r#"#[kube(schema = "{}")]"#, self.schema_mode)?;
+                    }
+
+                    for derive in &self.derive_traits {
+                        if derive.derived_trait == "JsonSchema" {
+                            continue;
+                        }
+
+                        if derive.derived_trait == "Default"
+                            && self.smart_derive_elision
+                            && !container.can_derive_default(&structs)
+                        {
+                            continue;
+                        }
+
+                        writeln!(&mut generated, r#"#[kube(derive="{}")]"#, derive.derived_trait)?;
+                    }
+                }
+
+                if container.is_enum {
+                    writeln!(&mut generated, "pub enum {} {{", container.name)?;
+                } else {
+                    writeln!(&mut generated, "pub struct {} {{", container.name)?;
+                }
+            } else {
+                self.write_derives(container, &structs, &mut generated)?;
+
+                let spec_trimmed_name = container.name.as_str().replace(
+                    &format!("{}Spec", kind.to_upper_camel_case()),
+                    &kind.to_upper_camel_case(),
+                );
+
+                if container.is_enum {
+                    writeln!(&mut generated, "pub enum {} {{", spec_trimmed_name)?;
+                } else {
+                    writeln!(&mut generated, "pub struct {} {{", spec_trimmed_name)?;
+                }
+            }
+
+            for member in &container.members {
+                self.write_docstr(&member.docs, "    ", &mut generated)?;
+
+                if !member.serde_annot.is_empty() {
+                    writeln!(&mut generated, "    #[serde({})]", member.serde_annot.join(", "))?;
+                }
+
+                let name = quote::format_ident!("{}", member.name);
+
+                for annotation in &member.extra_annot {
+                    writeln!(&mut generated, "    {}", annotation)?;
+                }
+
+                let spec_trimmed_type = member.type_.as_str().replace(
+                    &format!("{}Spec", kind.to_upper_camel_case()),
+                    &kind.to_upper_camel_case(),
+                );
+
+                if container.is_enum {
+                    // NB: only supporting plain enumerations atm, not oneOf
+                    writeln!(&mut generated, "    {},", name)?;
+                } else {
+                    writeln!(&mut generated, "    pub {}: {},", name, spec_trimmed_type)?;
+                }
+            }
+
+            writeln!(&mut generated, "}}")?;
+            writeln!(&mut generated)?;
+        }
+
+        Ok(generated)
```

Comment:
presumably most of this stuff is the same as in old main

Reply:
It's as identical as possible, basically just swapping out `println!(...)` for `writeln!(<buffer>, <whatever was already there>)?`